packages/enzyme-adapter-react-13/src/ReactThirteenAdapter.js

```diff
@@ -143,6 +143,7 @@ class ReactThirteenAdapter extends EnzymeAdapter {
     let isDOM = false;
     let cachedNode = null;
     return {
+      disableComponentDidUpdateOnSetState: true,
```

Comment:
This seems like a really awkward way to extend the adapter interface.

Reply:
What about this?

```js
return {
  options: {
    disableComponentDidUpdateOnSetState: true
  }
```

I can do this without to extend the adapter interface but it requires to access the React version directly from ShallowWrapper. I think ShallowWrapper should depend on adapters only.

Reply:
I wonder if the option should actually be the opposite. `enableComponentDidUpdateOnSetState`, making it an opt-in flag for adapters, and one that only react 16 will be using?

Reply:
@lelandrichardson I think ShallowRenderer won't support componentDidUpdate in future releases so the flag is required by only versions before v16. Using `enableComponentDidUpdateOnSetState` means that enzyme has to set the flag for future versions of React permanently. So I used `disableComponentDidUpdateOnSetState` for this.
But it might make sense to use `enableComponentDidUpdateOnSetState` and set the flag until dropping v15 support.

Reply:
that makes sense.  I think there might be more use cases for adding "options" for the adapter. I think i like giving them their own namespace though: `options.enableComponentDidUpdateOnSetState` or something.

Also, I wonder if it's better to put this on the adapter instance itself instead of the object it returns from `createRenderer`?  This way, we can set an empty `this.options = { ... }` in the `EnzymeAdapter` base constructor, which would mean people wouldn't have to set defaults.

Thoughts?

Reply:
That makes sense! I've updated this PR.